slatedb/src/db_snapshot.rs

```diff
@@ -121,6 +121,7 @@ impl DbSnapshot {
                 &db_state,
                 None,
                 Some(self.started_seq),
+                None,
```

Comment:
I want to make sure I understand why we don't include a range tracker here. If we have serializable snapshot isolation, don't we want to ensure read-write (snapshots) don't conflict?

Reply:
a range tracker tracks the read set for read-write conflict checks. my understanding is that it primarily serves for SSI transactions, where it helps detect if the read keys have been modified by other concurrent transactions. however, a `DbSnapshot` doesn't have a `commit()` operation but mostly a read-only consistent view, so i think we do not require range tracker here 🤔